### PR TITLE
Add integration testing for Product

### PR DESCRIPTION
**Pull request by trevormcdonald**
- Run tests with `docker-compose -f integration/docker-compose.yaml up`
- Create tests that call CLI functions and check their effects
- Test product create success
- Test product update success
- Test product delete success
- Add these tests to the `bin/run_integration_tests` script
- Add calls for CLI unit tests in `daemon/test/docker-compose.yaml` and `docker/tests`
Note: the unit tests are called by name to avoid also calling integration tests, which
will fail without all services running.
- Format docker compose yaml

This adds a separate docker-compose.yaml (`integration/docker-compose.yaml`) to run integration tests. The compose file is the same as the default `docker-compose.yaml` with logging turned down and a `cargo test` command to run the integration tests after starting the daemon. An optional `INTEGRATION_TEST_URL` can be provided, the default being `http://gridd:8080`.

The tests live in `cli/tests/product.rs`, according to Rust convention, and use the [assert_cmd](https://github.com/assert-rs/assert_cmd) crate recommended by [clap](https://github.com/clap-rs/clap#related-crates). `assert_cmd` takes the rust binary and runs it as a process with given args. The tests currently only verify that the commands do not return errors.

This also adds calls for unit tests in `daemon/test/docker-compose.yaml` and `docker/tests`.

See [Jira](https://jira.hyperledger.org/browse/GRID-106)
Depends on [PR 114](https://github.com/hyperledger/grid/pull/114) and [PR 115](https://github.com/hyperledger/grid/pull/115)

### CODE REVIEW

File: cli/tests/integration_tests.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
+extern crate assert_cmd;
```
Comment (agunde406) [inline cli/tests/integration_tests.rs]: missing license

File: cli/tests/integration_tests.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    static ORG_NAME: &str = "target";
+    static ORG_ADDRESS: &str = "target hq";
+
+    static PRODUCT_FILE: &str = "/build/cli/tests/test_product.yaml";
+
+    mod product {
+        use super::*;
+        #[test]
+        #[ignore]
+        fn test_product_create() {
```
Comment (agunde406) [inline cli/tests/integration_tests.rs]: Please add description to this test

File: cli/tests/integration_tests.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    static ORG_ID: &str = "314156";
+    static ORG_NAME: &str = "target";
+    static ORG_ADDRESS: &str = "target hq";
+
+    static PRODUCT_FILE: &str = "/build/cli/tests/test_product.yaml";
+
+    mod product {
+        use super::*;
+        #[test]
+        #[ignore]
```
Comment (agunde406) [inline cli/tests/integration_tests.rs]: Why is this ignored?

File: cli/tests/integration_tests.rs (hunk 1 above)
Comment (trevormcdonald) [reply cli/tests/integration_tests.rs]: 👍

File: cli/tests/integration_tests.rs (hunk 2 above)
Comment (trevormcdonald) [reply cli/tests/integration_tests.rs]: 👍

File: cli/tests/integration_tests.rs (hunk 3 above)
Comment (trevormcdonald) [reply cli/tests/integration_tests.rs]: 👍

File: daemon/test/docker-compose.yaml [hunk 4]
```diff
@@ -1,3 +1,5 @@
+--- # ------------------------------------------------------------------------------
```
Comment (vaporos) [inline daemon/test/docker-compose.yaml]: Was this intentional?

File: daemon/test/docker-compose.yaml [hunk 5]
```diff
@@ -11,39 +13,40 @@
...
-    test_server:
-        image: postgres
-        restart: always
-        ports:
-            - '5433:5432'
-        environment:
-            POSTGRES_USER: grid_test
-            POSTGRES_PASSWORD: grid_test
-            POSTGRES_DB: grid_test
+  test_server:
```
Comment (vaporos) [inline daemon/test/docker-compose.yaml]: Please move the file reformatting to a separate commit (it can still be in the same PR), so it is more obvious what the changes are vs. just reformatting. (Also provide justification for the reformatting in the commit message.)

File: daemon/test/docker-compose.yaml [hunk 6]
```diff
@@ -11,9 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# ------------------------------------------------------------------------------
```
Comment (agunde406) [inline daemon/test/docker-compose.yaml]: This line should not be removed in this commit. You fix it in the next commit.

File: daemon/test/docker-compose.yaml [hunk 7]
```diff
@@ -26,24 +24,27 @@ services:
...
+      bash -c "
+        # we need to wait for the db to have started.
+        until PGPASSWORD=grid_test psql -h test_server -U grid_test -c '\q'; do
+            >&2 echo \"Database is unavailable - sleeping\"
+            sleep 1
+        done
+        cd daemon
+        cargo test --features test-api --  --test-threads=1
+        cd ../cli
+        cargo test actions -- --test-threads=1
```
Comment (agunde406) [inline daemon/test/docker-compose.yaml]: Can you explain the additions of ` cargo test actions -- --test-threads=1` and ` cargo test yaml_parser -- --test-threads=1` in a commit message? Should this a part of the malformed commit?

File: integration/docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,269 @@
+--- # ------------------------------------------------------------------------------
```
Comment (agunde406) [inline integration/docker-compose.yaml]: You have an extra --- on the beginning of the line.

File: daemon/test/docker-compose.yaml (hunk 7 above)
Comment (trevormcdonald) [reply daemon/test/docker-compose.yaml]: I update the first commit message to include:

- Add these tests to the `bin/run_integration_tests` script
- Add calls for CLI unit tests in `daemon/test/docker-compose.yaml` and `docker/tests`
Note: the unit tests are called by name to avoid also calling integration tests, which
will fail without all services running.

I couldn't seem to find a better way to exclude the new tests with `cargo`.
